docs(cli): Setup from npm package (#9215)
redwoodjs/redwood: docs/docs/cli-commands.md

| `deploy`           | Set up a deployment configuration for a provider                                           |
| `generator`        | Copy default Redwood generator templates locally for customization                         |
| `i18n`             | Set up i18n                                                                                |

> | `package`          | Peform setup actions by running a third party npm package                                  |

Reviewer: ```suggestion
| `package`          | Peform setup actions by running a third-party npm package                                  |
```

> Note: To detect the RedwoodJS framework, please use netlify-cli v3.34.0 or greater.

### setup package

This command takes a published npm package that you specify, performs some compatibility checks, and then executes it's bin script. This allows you to use third-party packages that can provide you with an easy-to-use setup command for the particular functionality they provide.

This command behaves similarly to `yarn dlx` but will attempt to confirm compatibility between the package you are attempting to run and the current version of Redwood you are running. You can bypass this check by passing the `--force` flag if you feel you understand any potential compatibility issues.

```
yarn redwood setup package <npm-package>
```

| Arguments & Options | Description              |
| :------------------ | :----------------------- |

> | `--force, -f`       | Overwrite existing files |

Reviewer: ```suggestion
| `--force, -f`       | Forgo compatibility checks |
```

**Usage**

> Run the made up `@bluewoodjs/rw-setup` package:

Reviewer: Let's change the scope to `@redwoodjs` here so that we at least know what happens if someone doesn't really read these docs and just blindly copy-pastes this codeblock:

```suggestion
Run the made up `@redwoodjs/setup-example` package:
```

```bash

> ~/redwood-app$ yarn rw setup package @bluewoodjs/rw-setup

Reviewer: ```suggestion
~/redwood-app$ yarn rw setup package @redwoodjs/setup-example
```

```

Run the same package but using a particular npm tag and avoiding any compatibility checks:
```bash

> ~/redwood-app$ yarn rw setup package @bluewoodjs/rw-setup@beta --force

Reviewer: ```suggestion
~/redwood-app$ yarn rw setup package @redwoodjs/setup-example@beta --force
```

```

**Compatibility Checks**

> We perform a simple compatibility check in an attempt to make you aware of potential compatibility issues with setup packages you might wish to run. This works by examining the version of `@redwoodjs/core` you are using within your root `package.json`. We compare this value with a compatibility range the npm package specifies in the `engines.redwoodjs` field of its own `package.json`. If the version of `@redwoodjs/core` you are using falls outside of the compatibility range specified by the package you are attempting to run, we will warn you and ask you to confirm you wish to continue.

Reviewer: ```suggestion
We perform a simple compatibility check in an attempt to make you aware of potential compatibility issues with setup packages you might wish to run. This works by examining the version of `@redwoodjs/core` you are using within your root `package.json`. We compare this value with a compatibility range the npm package specifies in the `engines.redwoodjs` field of its own `package.json`. If the version of `@redwoodjs/core` you are using falls outside of the compatibility range specified by the package you are attempting to run, we will warn you and ask you to confirm that you wish to continue.
```

> It's the responsibility of the authors of those npm packages to specify the correct range of compatibility so you should always research the packages you use with this command. Especially since they will be executing code on your machine!

Reviewer: ```suggestion
It's the author of the npm package's responsibility to specify the correct compatibility range, so **you should always research the packages you use with this command**. Especially since they will be executing code on your machine!
```
